app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :name, presence: true, length: {maximum: Settings.valid.name.maxlength}
+  validates :email, presence: true, length: {minimum: Settings.valid.email.minlength, maximum: Settings.valid.email.maxlength},
+    format: {with: Settings.regex.VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  validates :password, presence: true, length: {minimum: Settings.valid.password.minlength}, if: :password
+  has_secure_password
```

Comment:
`has_secure_password` -> thằng này dùng để làm gì đây em?

Reply:
thằng này ngầm định sẽ có 3 trường là password , password_confirmation, password_digest trong model User, khi 1 instance user sẽ so sánh giữa password và password_confirmation nếu bằng nhau thì sẽ hash password bằng 1 thuật toán mã hóa trong gem bcrypt và khi save thì chỉ password_digest đcược lưu trong db